test/humanStandardToken.js

```diff
@@ -1,345 +1,186 @@
-      ctr = result
-      return ctr.transfer.call(accounts[1], 0, {from: accounts[0]})
-    }).then(function (result) {
-      assert.isTrue(result)
-    }).catch((err) => { throw new Error(err) })
-  })
-
-    // NOTE: testing uint256 wrapping is impossible in this standard token since you can't supply > 2^256 -1.
-
-    // todo: transfer max amounts.
-
-// APPROVALS
-
-  it('approvals: msg.sender should approve 100 to accounts[1]', function () {
-    var ctr = null
-    return HumanStandardToken.new(10000, 'Simon Bucks', 1, 'SBX', {from: accounts[0]}).then(function (result) {
-      ctr = result
-      return ctr.approve(accounts[1], 100, {from: accounts[0]})
-    }).then(function (result) {
-      return ctr.allowance.call(accounts[0], accounts[1])
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 100)
-    }).catch((err) => { throw new Error(err) })
-  })
-
-  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient. It should succeed.', function () {
-    var ctr = null
-    var sampleCtr = null
-    return HumanStandardToken.new(10000, 'Simon Bucks', 1, 'SBX', {from: accounts[0]}).then(function (result) {
-      ctr = result
-      return SampleRecipientSuccess.new({from: accounts[0]})
-    }).then(function (result) {
-      sampleCtr = result
-      return ctr.approveAndCall(sampleCtr.address, 100, '0x42', {from: accounts[0]})
-    }).then(function (result) {
-      return ctr.allowance.call(accounts[0], sampleCtr.address)
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 100)
-      return sampleCtr.value.call()
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 100)
-    }).catch((err) => { throw new Error(err) })
-  })
+  // TRANSERS
+  // normal transfers without approvals.
 
-  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient and throw.', function () {
-    var ctr = null
-    var sampleCtr = null
-    return HumanStandardToken.new(10000, 'Simon Bucks', 1, 'SBX', {from: accounts[0]}).then(function (result) {
-      ctr = result
-      return SampleRecipientThrow.new({from: accounts[0]})
-    }).then(function (result) {
-      sampleCtr = result
-      return ctr.approveAndCall.call(sampleCtr.address, 100, '0x42', {from: accounts[0]})
-    }).catch(function (result) {
-            // It will catch OOG.
+  // this is not *good* enough as the contract could still throw an error otherwise.
+  // ideally one should check balances before and after, but estimateGas currently always throws an error.
+  // it's not giving estimate on gas used in the event of an error.
+  it('transfers: ether transfer should be reversed.', async () => {
+    try {
+      web3.eth.sendTransaction({from: accounts[0], to: HST.address, value: web3.toWei('10', 'Ether')})
+    } catch (e) {
```

Comment:
I realize this was not in the in the pre-existing tests, but please check that the balance of the contract is unchanged, then I think we can remove the above comment.

Reply:
Also, in this catch block, using your new utils.js, this could check for a throw via invalid opcode.

Reply:
Yep, while working on another test suite, I found out the error you're getting is probably more like:

{ message: 'Web3ProviderEngine does not support synchronous requests.',
  stack: 'Error: Web3ProviderEngine does not support synchro…/truffle/node_modules/mocha/lib/runner.js:443:10)',
  name: 'ProviderError' }
> e.message.search('invalid
```{ message: 'Web3ProviderEngine does not support synchronous requests.',
  stack: 'Error: Web3ProviderEngine does not support synchro…/truffle/node_modules/mocha/lib/runner.js:443:10)',
  name: 'ProviderError' }```

Reply:
I think there is a misunderstanding here, I didn't wrote this test, it was there before : https://github.com/ConsenSys/Tokens/blob/staging/test/humanStandardToken.js#L55

Reply:
I had to use `expectThrow` into the sendTransaction callback as this call doesn't return a promise